src/gui/guiFormSpecMenu.cpp

```diff
@@ -681,8 +835,47 @@ void GUIFormSpecMenu::parseBackground(parserData* data, const std::string &eleme
 		if (!data->explicit_size && !clip)
 			warningstream << "invalid use of unclipped background without a size[] element" << std::endl;
 
-		m_backgrounds.emplace_back(name, pos, geom, clip);
+		video::ITexture *texture = m_tsrc->getTexture(name);
+		if (texture != nullptr) {
+			FieldSpec spec(
+				name,
+				L"",
+				L"",
+				258+m_fields.size()
+			);
+
+			gui::IGUIImage *e = nullptr;
+			if (!clip) {
+				core::rect<s32> rect(pos, pos + geom);
+				e = Environment->addImage(rect, current_parent,
+						spec.fid, 0, true);
+				if (current_parent == this)
+					e->setNotClipped(true);
+			} else if (current_parent == this) {
+				core::rect<s32> rect(-pos, AbsoluteRect.getSize() + pos);
+				e = Environment->addImage(rect, current_parent,
+						spec.fid, 0, true);
+				e->setNotClipped(true);
+			} else {
+				// clipped to scroll_container => clip to clipper rect
+				const core::rect<s32> &window_rect =
+						current_parent->getParent()->getRelativePosition();
+				core::rect<s32> rect(window_rect.UpperLeftCorner - pos,
+						window_rect.LowerRightCorner + pos);
+				e = Environment->addImage(rect,
+						current_parent->getParent()->getParent(), spec.fid, 0, true);
+			}
+
+			e->setImage(texture);
```

Comment:
For safety, please add a `nullptr` check here (`FATAL_ERROR_IF` or similar)

Reply:
There's already a `if (texture != nullptr) {` before, so I guess, this is about `e`.

Reply:
Done.

Reply:
Yes, I meant `e`. Just in case the code is either touched or Irrlicht cannot add the image for some reason.